### PR TITLE
Python. Homework #1. Task #1. Куковеров Егор

### PR DESCRIPTION
**Pull request by ubuntuegor**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "python.formatting.provider": "black",
+  "python.formatting.blackArgs": [
+    "-l",
+    "120"
+  ],
+  "editor.formatOnSave": true
+}
```
Comment (areyde) [inline .vscode/settings.json]: Are you sure you need to push this file to the repo?

File: src/hw1/head.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import argparse
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-n", dest="lines", type=int, default=10, help="Number of displayed lines.")
+    parser.add_argument("file", help="File to read.")
+    args = parser.parse_args()
+
+    with open(args.file, "r") as f:
```
Comment (areyde) [inline src/hw1/head.py]: You don't need to write `r`, it's the default behavior.

Comment (areyde) [reply src/hw1/head.py]: (everywhere)

File: src/hw1/nl.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("file", help="File to read.")
+    args = parser.parse_args()
+
+    with open(args.file, "r") as f:
+        for i, line in enumerate(f):
+            print(f"{i + 1} {line}", end="")
```
Comment (areyde) [inline src/hw1/nl.py]: `nl` does not enumerate emptly lines.

File: test/hw1/test_nl.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+def test_nl(tmp_path: pathlib.Path, capsys: pytest.CaptureFixture):
+    test_content = """black==22.1.0
+mypy==0.931
+pytest==7.0.1
+"""
+    result_content = """1 black==22.1.0
+2 mypy==0.931
+3 pytest==7.0.1
+"""
```
Comment (areyde) [inline test/hw1/test_nl.py]: Add a test with empty lines for it to be correct.

**Review by areyde — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (ubuntuegor) [reply .vscode/settings.json]: Yes! It provides automatic formatting for my editor. It's the same code formatter that we use in our CI, and we can run it every now and then rather than fix everything manually. Who wouldn't want that? I might agree this is not the best PR for it though.

**Review by areyde — APPROVED** [review]
🌮
